src/www/app/outline_server.ts

```diff
@@ -88,10 +100,241 @@ export class OutlineServer implements PersistentServer {
   }
 
   checkReachable(): Promise<boolean> {
-    return this.tunnel.isReachable();
+    return this.tunnel.isReachable(this.config);
   }
 
-  public static isServerCipherSupported(cipher?: string) {
+  static isServerCipherSupported(cipher?: string) {
     return cipher !== undefined && OutlineServer.SUPPORTED_CIPHERS.includes(cipher);
   }
 }
+
+// DEPRECATED: V0 server persistence format.
+export interface ConfigById {
+  [serverId: string]: ShadowsocksConfig;
+}
+
+// V1 server persistence format.
+export interface AccessKeyById {
```

Comment:
A dictionary is not a very convenient format, because the type is weird. It's not a fixed schema, since you have variable fields. It's better to use a list.

Also, it seems the access key is not all we need.

I think we need something like:
```ts
type AccessKey = string;

interface ServerJson {
  id: string
  name?:  string
  accessKey: AccessKey  // device key for connection. Whatever you get from the permanent redirects.
  sharedAccessKey?: AccessKey  // multi-user key for sharing
}
```
`accessKey` could be name `connectionAccessKey`.
`sharedAccessKey`  could have other names, such as `originalAccessKey`.

The is_outline bit could be derived from the sharedAccessKey,  or accessKey if absent.

The localStorage format under `servers_v1` would be `ServerJson[]`.

Reply:
Nice suggestion. Done in bc68e17 (without multi-user keys).